sstables/sstables_manager.hh

```diff
@@ -91,6 +92,10 @@ public:
 
     std::unique_ptr<sstable_directory::components_lister> get_components_lister(const data_dictionary::storage_options& storage, std::filesystem::path dir);
 
+    s3::endpoint_config_ptr get_endpoint_config(sstring endpoint) const {
+        return _object_storage_config.at(endpoint);
+    }
```

Comment:
Shouldn't it still work without the endpoint configuration file? 

If not, I suggest explicitly checking for the existence of endpoint at map and dumping a meaningful error message explaining that the endpoint wasn't configured through the config file.

Otherwise, someone playing with s3 at local instance, without a config file specified, would be confused at cryptic error messages that could happen when creating a new sstable object which calls get_endpoint_config().

Reply:
Fair point. But I guess it's too late to abort here -- when sstables are created for the 1st time. I think there are two better options.

First -- check the config when S3-backed keyspace is created and fail creation if config is missing
Second -- apply default config for missing endpoints, i.e. plain http without key/secret and port 80

The 2nd option looks troublesome because if the endpoint is not configured sstables creation would succeed, but _flushing_ would fail. On the other hand, the 1st option would fail similarly if the endpoint config exists, but is in any sense invalid (e.g. -- misprint in port/key/whatever).

Reply:
Perhaps we can sanity test the endpoint configuration (default, user-feeded, whatever) when creating its instance (s3::endpoint_config)? Like a simple file_output_stream -> flush.

So we also catch the problem where the endpoint config info become later outdated. It seems to me checking it only when creating the keyspace is not enough. Note it can also become outdated while the Scylla is running, so we should catch this scenario and dump a meaningful error message advicing the user that intervention is needed.

Should we abort Scylla if an error in the config is caught? Probably yes, because the replica will be unable to serve the workload on s3-backed table.